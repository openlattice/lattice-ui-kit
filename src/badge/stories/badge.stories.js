import React from 'react';
import { storiesOf } from '@storybook/react';

import { Card, CardSegment } from '../../layout';
import Badge from '..';

storiesOf('Badge', module)
  .add('default', () => (
    <>
      <h1>Badges</h1>
      <Card>
        <CardSegment>
          <Badge>5</Badge>
          <Badge mode="primary">25</Badge>
          <Badge mode="secondary">10</Badge>
          <Badge mode="subtle">25</Badge>
          <Badge mode="added">+100</Badge>
          <Badge mode="removed">-99</Badge>
        </CardSegment>
        <CardSegment>
          <Badge mode="primary" max="100" />
        </CardSegment>
      </Card>
    </>
  ));
