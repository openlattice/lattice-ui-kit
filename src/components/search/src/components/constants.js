import { List, Map, fromJS } from 'immutable';

const mockFilterFields = [
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'reportType',
    label: 'Report type',
    options: ['Crisis Template', 'Follow-up'],
  },
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'badges',
    label: 'Badges',
    options: ['Officer Safety', 'Substance use'],
  },
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'submitter',
    label: 'Submitter',
    options: [
      'squidward@bubblebowl.com',
      'smitty@werbenjagermanjensen.com',
      'wumbology@thestudyofwumbo.com',
      'enigma@innermachinations.com'
    ],
  },
];

const mockSearchResultsForReports = List([
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety'],
    submitter: 'squidward@bubblebowl.com'
  }),
  Map({
    reportType: 'Follow-up',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'squidward@bubblebowl.com',
  }),
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'smitty@werbenjagermanjensen.com'
  }),
  Map({
    reportType: 'Follow-up',
    badges: ['Substance use'],
    submitter: 'wumbology@thestudyofwumbo.com'
  }),
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'enigma@innermachinations.com'
  }),
  Map({
    reportType: 'Follow-up',
    submitter: 'enigma@innermachinations.com'
  }),
]);

const mockSearchResultsForPeople = List([
  Map({
    lastName: 'Werbenjagermanjensen',
    firstName: 'Smitty',
    middleName: '',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Fish',
    dob: '02/22/2002',
    identifier: '365ba027-0c71-47d1-a5f4-61346ed96d8c',
  }),
  Map({
    lastName: 'Thornton',
    firstName: 'Cameron',
    middleName: 'Orville',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Black / African American',
    dob: '01/05/1982',
    identifier: '565ba027-0c71-47d1-a5f4-61346ed96d8c',
  }),
  Map({
    lastName: 'Gerner',
    firstName: 'Timothy',
    middleName: 'Pernell',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Native Hawaiian or Other Pacific Islander',
    dob: '10/02/1991',
    identifier: '465ba027-0c71-47d1-a5f4-61346ed96d8c',
  })
]);

export {
  mockFilterFields,
  mockSearchResultsForReports,
  mockSearchResultsForPeople
};
