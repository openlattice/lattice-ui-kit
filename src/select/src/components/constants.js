export const LATTICE_SELECT = 'lattice-select';
export const OPTIONS = [
  { label: '0', value: 'zero' },
  { label: '1', value: 'one' },
  { label: '2', value: 'two' },
  { label: '3', value: 'three' },
  { label: '4', value: 'four' },
];
