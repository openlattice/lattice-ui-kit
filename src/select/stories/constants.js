// @flow

/* eslint-disable max-len */
const REACT_SELECT_USAGE :string = 'Select and Creatable components are simple wrappers that pass OpenLattice styling to their respective react-select components. All props provided to these wrappers are passed through allowing full use of react-select functionality.';

const stateOptions = [
  { value: 'AL', label: 'Alabama' },
  { value: 'AK', label: 'Alaska' },
  { value: 'AS', label: 'American Samoa' },
  { value: 'AZ', label: 'Arizona' },
  { value: 'AR', label: 'Arkansas' },
  { value: 'CA', label: 'California' },
  { value: 'CO', label: 'Colorado' },
  { value: 'CT', label: 'Connecticut' },
  { value: 'DE', label: 'Delaware' },
  { value: 'DC', label: 'District Of Columbia' },
  { value: 'FM', label: 'Federated States Of Micronesia' },
  { value: 'FL', label: 'Florida' },
  { value: 'GA', label: 'Georgia' },
  { value: 'GU', label: 'Guam' },
  { value: 'HI', label: 'Hawaii' },
  { value: 'ID', label: 'Idaho' },
  { value: 'IL', label: 'Illinois' },
  { value: 'IN', label: 'Indiana' },
  { value: 'IA', label: 'Iowa' },
  { value: 'KS', label: 'Kansas' },
  { value: 'KY', label: 'Kentucky' },
  { value: 'LA', label: 'Louisiana' },
  { value: 'ME', label: 'Maine' },
  { value: 'MH', label: 'Marshall Islands' },
  { value: 'MD', label: 'Maryland' },
  { value: 'MA', label: 'Massachusetts' },
  { value: 'MI', label: 'Michigan' },
  { value: 'MN', label: 'Minnesota' },
  { value: 'MS', label: 'Mississippi' },
  { value: 'MO', label: 'Missouri' },
  { value: 'MT', label: 'Montana' },
  { value: 'NE', label: 'Nebraska' },
  { value: 'NV', label: 'Nevada' },
  { value: 'NH', label: 'New Hampshire' },
  { value: 'NJ', label: 'New Jersey' },
  { value: 'NM', label: 'New Mexico' },
  { value: 'NY', label: 'New York' },
  { value: 'NC', label: 'North Carolina' },
  { value: 'ND', label: 'North Dakota' },
  { value: 'MP', label: 'Northern Mariana Islands' },
  { value: 'OH', label: 'Ohio' },
  { value: 'OK', label: 'Oklahoma' },
  { value: 'OR', label: 'Oregon' },
  { value: 'PW', label: 'Palau' },
  { value: 'PA', label: 'Pennsylvania' },
  { value: 'PR', label: 'Puerto Rico' },
  { value: 'RI', label: 'Rhode Island' },
  { value: 'SC', label: 'South Carolina' },
  { value: 'SD', label: 'South Dakota' },
  { value: 'TN', label: 'Tennessee' },
  { value: 'TX', label: 'Texas' },
  { value: 'UT', label: 'Utah' },
  { value: 'VT', label: 'Vermont' },
  { value: 'VI', label: 'Virgin Islands' },
  { value: 'VA', label: 'Virginia' },
  { value: 'WA', label: 'Washington' },
  { value: 'WV', label: 'West Virginia' },
  { value: 'WI', label: 'Wisconsin' },
  { value: 'WY', label: 'Wyoming' },
];

const paragraph = `
October arrived, spreading a damp chill over the grounds and into the castle. Madam Pomfrey, the nurse, was kept busy by a sudden spate of colds among the staff and students. Her Pepperup potion worked instantly, though it left the drinker smoking at the ears for several hours afterward. Ginny Weasley, who had been looking pale, was bullied into taking some by Percy. The steam pouring from under her vivid hair gave the impression that her whole head was on fire. Raindrops the size of bullets thundered on the castle windows for days on end; the lake rose, the flower beds turned into muddy streams, and Hagrid's pumpkins swelled to the size of garden sheds. Oliver Wood's enthusiasm for regular training sessions, however, was not dampened, which was why Harry was to be found, late one stormy Saturday afternoon a few days before Halloween, returning to Gryffindor Tower, drenched to the skin and splattered with mud.
`;

export {
  REACT_SELECT_USAGE,
  paragraph,
  stateOptions,
};
