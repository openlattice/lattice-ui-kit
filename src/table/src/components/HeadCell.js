// @flow
import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import type { Node } from 'react';
import { faSort, faSortUp, faSortDown } from '@fortawesome/pro-duotone-svg-icons';

import { Cell } from './styled';
import type { SortOrder } from '../../types';

type Props = {
  cellStyle ? :Object;
  children :Node;
  className ? :string;
  onClick ? :(e :SyntheticEvent<HTMLElement>) => void;
  order ? :SortOrder;
  sortable ? :boolean;
};

const HeadCell = (props :Props) => {
  const {
    cellStyle,
    children,
    className,
    onClick,
    order,
    sortable,
  } = props;

  let icon = faSort;
  if (order === 'asc') icon = faSortUp;
  if (order === 'desc') icon = faSortDown;

  const sortText = `Sort (${order === 'asc' ? 'Ascending' : 'Descending'})`;

  return (
    <Cell
        as="th"
        cellStyle={cellStyle}
        className={className}
        onClick={onClick}>
      {children}
      { (sortable) && <span aria-label={sortText}><FontAwesomeIcon icon={icon} fixedWidth /></span> }
    </Cell>
  );

};

HeadCell.defaultProps = {
  cellStyle: {},
  className: undefined,
  onClick: undefined,
  order: false,
  sortable: false,
};

export default HeadCell;
