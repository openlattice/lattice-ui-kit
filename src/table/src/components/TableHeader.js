// @flow
import React from 'react';
import isFunction from 'lodash/isFunction';

import { StyledRow } from './styled';
import type { SortOrder } from '../../types';

type Props = {
  className ? :string;
  components :Object;
  headers ? :Object[];
  onSort ? :(event :SyntheticEvent<HTMLElement>, property :string) => void;
  order ? :SortOrder;
  orderBy ? :string;
  sticky ? :boolean;
};


const TableHeader = (props :Props) => {
  const {
    components,
    className,
    headers,
    onSort,
    order,
    orderBy,
    sticky,
  } = props;

  const createSortHandler = (property :string) => (event :SyntheticEvent<HTMLElement>) => {
    if (isFunction(onSort)) {
      onSort(event, property);
    }
  };

  return (
    <thead className={className}>
      <StyledRow sticky={sticky}>
        {
          headers && headers.map((header) => {
            const { key, label, cellStyle } = header;
            return (
              <components.HeadCell
                  key={key}
                  components={components}
                  cellStyle={cellStyle}
                  onClick={onSort ? createSortHandler(key) : undefined}
                  order={orderBy === header.key ? order : false}
                  sortable>
                {label}
              </components.HeadCell>
            );
          })
        }
      </StyledRow>
    </thead>
  );
};

TableHeader.defaultProps = {
  className: undefined,
  headers: [],
  onSort: undefined,
  order: false,
  orderBy: undefined,
  sticky: true,
};

export default TableHeader;
