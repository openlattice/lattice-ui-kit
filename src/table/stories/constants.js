const TABLE_HEADERS = [
  { key: 'name', label: 'Subject Name', cellStyle: { width: '50%' } },
  { key: 'dob', label: 'Date of Birth' },
  { key: 'manager', label: 'Manager' },
  { key: 'lastUpdated', label: 'Last Updated' },
  { key: 'id', label: 'ID', sortable: false },
];

const TABLE_DATA = [
  {
    name: 'Eugene Krabs',
    dob: '1942-11-30',
    manager: 'Spongebob',
    lastUpdated: '2019-09-01',
    id: '1'
  },
  {
    name: 'Squidward Tentacles',
    dob: '',
    manager: 'Smitty',
    lastUpdated: '2019-08-29',
    id: '2'
  },
  {
    name: 'Sandy Cheeks',
    dob: '1987-11-17',
    manager: 'Patrick',
    lastUpdated: '2019-07-30',
    id: '3'
  },
  {
    name: 'Larry the Lobster',
    dob: '1975-06-23',
    manager: 'Patchy',
    lastUpdated: '2019-08-16',
    id: '4'
  },
  {
    name: 'Sheldon Plankton',
    dob: '1942-11-30',
    manager: 'Spongebob',
    lastUpdated: '2019-08-20',
    id: '5'
  },
  {
    name: 'Mrs. Puff',
    dob: '',
    manager: 'Spongebob',
    lastUpdated: '2019-08-21',
    id: '6'
  },
  {
    name: 'Flying Dutchman',
    dob: '1678-04-20',
    manager: 'Spongebob',
    lastUpdated: '2019-06-21',
    id: '7'
  },
];

export {
  TABLE_HEADERS,
  TABLE_DATA,
};
