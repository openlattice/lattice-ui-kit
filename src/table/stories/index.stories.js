import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';

import { Card, CardSegment } from '../../layout';
import { TABLE_DATA, TABLE_HEADERS } from './constants';
import { getHoverStyles } from '../../utils/StyleUtils';
import {
  Table
} from '..';

const CustomRow = styled.tr`
  ${getHoverStyles};
`;

const components = {
  Row: () => (
    <CustomRow onClick={() => {}}>
      <td colSpan="4">row</td>
    </CustomRow>
  )
};

storiesOf('Table', module)
  .add('Sortable', () => (
    <Card>
      <CardSegment vertical>
        <Table
            headers={TABLE_HEADERS}
            data={TABLE_DATA} />
      </CardSegment>
    </Card>
  ))
  .add('Paginated', () => (
    <Card>
      <CardSegment vertical>
        <Table
            headers={TABLE_HEADERS}
            data={TABLE_DATA}
            rowsPerPageOptions={[5, 20, 50]}
            paginated />
      </CardSegment>
    </Card>
  ))
  .add('Custom Components', () => (
    <Card>
      <CardSegment vertical>
        <Table
            components={components}
            headers={TABLE_HEADERS}
            data={TABLE_DATA}
            rowsPerPageOptions={[5, 20, 50]}
            paginated />
      </CardSegment>
    </Card>
  ));
