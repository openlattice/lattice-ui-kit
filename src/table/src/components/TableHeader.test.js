import React from 'react';
import { shallow } from 'enzyme';

import TableHeader from './TableHeader';
import { TABLE_HEADERS } from '../../stories/constants';
import HeadCell from './HeadCell';

describe('TableHeader', () => {

  describe('props', () => {
    describe('sticky', () => {
      test('pass sticky prop to TableRow', () => {
        const wrapper = shallow(<TableHeader sticky />);

        const tableRowWrapper = wrapper.find('TableRow');
        expect(tableRowWrapper.prop('sticky')).toEqual(true);
      });
    });

    describe('onSort', () => {
      test('invoke onSort with property when HeadCell is clicked', () => {
        const mockOnSort = jest.fn();
        const wrapper = shallow(<TableHeader onSort={mockOnSort} headers={TABLE_HEADERS} />);

        const headerCells = wrapper.find(HeadCell);
        expect(mockOnSort).toHaveBeenCalledTimes(0);

        headerCells.get(0).props.onClick();
        expect(mockOnSort).toHaveBeenCalledTimes(1);
        expect(mockOnSort.mock.calls[0][1]).toEqual('name');

        headerCells.get(1).props.onClick();
        expect(mockOnSort).toHaveBeenCalledTimes(2);
        expect(mockOnSort.mock.calls[1][1]).toEqual('dob');
      });
    });

    describe('order and orderBy', () => {
      test('should pass order to HeadCell when orderBy matches header key', () => {
        const wrapper = shallow(<TableHeader headers={TABLE_HEADERS} order="asc" orderBy="name" />);

        const headerCells = wrapper.find(HeadCell);
        expect(headerCells.get(0).props.order).toEqual('asc');
        expect(headerCells.get(1).props.order).toEqual(false);
        expect(headerCells.get(2).props.order).toEqual(false);
        expect(headerCells.get(3).props.order).toEqual(false);
      });
    });
  });

  describe('render', () => {
    test('should render a thead at the root', () => {
      const wrapper = shallow(<TableHeader />);
      expect(wrapper.type()).toEqual('thead');
    });

    test('should render sortable HeadCells', () => {
      const wrapper = shallow(<TableHeader headers={TABLE_HEADERS} />);
      const headerCells = wrapper.find(HeadCell);

      headerCells.forEach((headerCell) => {
        expect(headerCell.prop('sortable')).toEqual(true);
      });
      expect(headerCells).toHaveLength(TABLE_HEADERS.length);
    });
  });
});
