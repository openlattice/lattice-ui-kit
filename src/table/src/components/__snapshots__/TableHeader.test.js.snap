// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader props onSort invoke onSort with property when HeadCell is clicked 1`] = `
<thead>
  <styled__StyledRow
    sticky={true}
  >
    <HeadCell
      cellStyle={
        Object {
          "width": "50%",
        }
      }
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="name"
      onClick={[Function]}
      order={false}
      sortable={true}
    >
      Subject Name
    </HeadCell>
    <HeadCell
      cellStyle={Object {}}
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="dob"
      onClick={[Function]}
      order={false}
      sortable={true}
    >
      Date of Birth
    </HeadCell>
    <HeadCell
      cellStyle={Object {}}
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="manager"
      onClick={[Function]}
      order={false}
      sortable={true}
    >
      Manager
    </HeadCell>
    <HeadCell
      cellStyle={Object {}}
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="lastUpdated"
      onClick={[Function]}
      order={false}
      sortable={true}
    >
      Last Updated
    </HeadCell>
  </styled__StyledRow>
</thead>
`;

exports[`TableHeader props order and orderBy should pass order to HeadCell when orderBy matches header key 1`] = `
<thead>
  <styled__StyledRow
    sticky={true}
  >
    <HeadCell
      cellStyle={
        Object {
          "width": "50%",
        }
      }
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="name"
      order="asc"
      sortable={true}
    >
      Subject Name
    </HeadCell>
    <HeadCell
      cellStyle={Object {}}
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="dob"
      order={false}
      sortable={true}
    >
      Date of Birth
    </HeadCell>
    <HeadCell
      cellStyle={Object {}}
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="manager"
      order={false}
      sortable={true}
    >
      Manager
    </HeadCell>
    <HeadCell
      cellStyle={Object {}}
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="lastUpdated"
      order={false}
      sortable={true}
    >
      Last Updated
    </HeadCell>
  </styled__StyledRow>
</thead>
`;

exports[`TableHeader render should render a thead at the root 1`] = `
<thead>
  <styled__StyledRow
    sticky={true}
  />
</thead>
`;

exports[`TableHeader render should render sortable HeadCells 1`] = `
<thead>
  <styled__StyledRow
    sticky={true}
  >
    <HeadCell
      cellStyle={
        Object {
          "width": "50%",
        }
      }
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="name"
      order={false}
      sortable={true}
    >
      Subject Name
    </HeadCell>
    <HeadCell
      cellStyle={Object {}}
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="dob"
      order={false}
      sortable={true}
    >
      Date of Birth
    </HeadCell>
    <HeadCell
      cellStyle={Object {}}
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="manager"
      order={false}
      sortable={true}
    >
      Manager
    </HeadCell>
    <HeadCell
      cellStyle={Object {}}
      components={
        Object {
          "HeadCell": [Function],
        }
      }
      key="lastUpdated"
      order={false}
      sortable={true}
    >
      Last Updated
    </HeadCell>
  </styled__StyledRow>
</thead>
`;
